src/core/stream/buffers_handler.ts

```diff
@@ -403,9 +419,25 @@ export default function BuffersHandler(
             })
         );
 
+    const lastPeriodBuffer$ = currentBuffer$
+      .filter((evt) => {
+        return evt.type === "periodBufferReady" || evt.type === "representationChange";
+      })
+      .switchMap(() => {
+        return currentBuffer$
+          .filter((evt) => evt.type === "full")
+          .take(1);
+      })
+      .withLatestFrom(lastBufferCreated$.take(1))
+      .mapTo({
+        type: "last-full",
+        value : bufferType,
+      });
```

Comment:
Not a fan of that ``lastBufferCreated$`` thing.

In reality, it's more along the line of the last period of a "finished" Manifest.

May be a Manifest method (like getLastPeriod?) or something else.

The problem here is that we rely to much on our current logic of buffer creation. The time at which the last buffer is created could change and that's not intrinsically linked to the last period.